src/Catalog/Monitoring/ValidationTestCollector.cs

```diff
@@ -0,0 +1,78 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Newtonsoft.Json.Linq;
+using NuGet.Services.Metadata.Catalog.Monitoring.Validation.Result;
+using NuGet.Services.Metadata.Catalog.Monitoring.Validation.Test;
+using NuGet.Services.Metadata.Catalog.Persistence;
+using Microsoft.Extensions.Logging;
+using NuGet.Services.Metadata.Catalog.Helpers;
+
+namespace NuGet.Services.Metadata.Catalog.Monitoring
+{
+    /// <summary>
+    /// Runs a <see cref="PackageValidationTestRunner"/> against collected packages.
+    /// </summary>
+    public class ValidationTestCollector : SortingIdVersionCollector
+    {
+        public ValidationTestCollector(Storage auditingStorage, PackageValidationTestRunner packageValidationTestRunner, Uri index, Func<HttpMessageHandler> handlerFunc = null, ILoggerFactory loggerFactory = null) : base(index, handlerFunc)
+        {
+            _auditingStorage = auditingStorage;
+            _packageValidationTestRunner = packageValidationTestRunner;
+            _loggerFactory = loggerFactory ?? throw new ArgumentNullException(nameof(loggerFactory));
+            _logger = loggerFactory?.CreateLogger<ValidationTestCollector>();
+        }
+
+        protected override async Task ProcessSortedBatch(CollectorHttpClient client, KeyValuePair<FeedHelpers.FeedPackageIdentity, IList<JObject>> sortedBatch, JToken context, CancellationToken cancellationToken)
+        {
+            var packageId = sortedBatch.Key.Id;
+            var packageVersion = sortedBatch.Key.Version;
+            var catalogEntriesJson = sortedBatch.Value;
+
+            try
+            {
+                var testResults = await _packageValidationTestRunner.Run(packageId, packageVersion, catalogEntriesJson, _auditingStorage, client, cancellationToken, _loggerFactory);
+                _logger.LogInformation("Finished testing {PackageId} {PackageVersion}", packageId, packageVersion);
+                var groupedResults = testResults.EndpointValidationResults.SelectMany(r => r.ValidationResults).GroupBy(r => r.Result);
+                foreach (var resultsWithResult in groupedResults)
+                {
+                    foreach (var validationTestResult in resultsWithResult)
+                    {
+                        var testResultLogString = "{PackageId} {PackageVersion}: {ValidationTest}: {TestResult}";
+                        var validationTestName = validationTestResult.ValidationTest.GetType().Name;
+                        var testResultString = validationTestResult.Result.ToString();
+
+                        switch (validationTestResult.Result)
+                        {
+                            case TestResult.Pass:
+                                _logger.LogInformation(testResultLogString, packageId, packageVersion, validationTestName, testResultString);
+                                break;
+                            case TestResult.Skip:
```

Comment:
Initially I had `Skip` logging a warning instead of `Information`, but @joelverhagen suggested against it. Any thoughts?